apps/frontend/core-dapp/src/features/withdraw/WithdrawStore.ts

```diff
@@ -264,21 +268,30 @@ export class WithdrawStore {
     if (withdrawalAmountInWstEthBN === undefined) return
 
     try {
-      const withdrawalAmountInEthBN = await this.root.balancerStore.quoteWstEthAmountInEth(
+      const withdrawalAmountInEth = await this.root.balancerStore.quoteWstEthAmountInEth(
         withdrawalAmountInWstEthBN
       )
 
-      if (withdrawalAmountInEthBN !== undefined) {
+      if (withdrawalAmountInEth !== undefined) {
         runInAction(() => {
-          this.withdrawalMarketValueInEth = { status: 'queried', value: withdrawalAmountInEthBN }
+          if (
+            this.root.advancedSettingsStore.isPriceImpactTooHigh(withdrawalAmountInEth.priceImpact)
+          ) {
+            this.withdrawalMarketValueInEth = { status: 'price-impact-too-high' }
+          } else {
+            this.withdrawalMarketValueInEth = {
+              status: 'queried',
+              value: withdrawalAmountInEth.value,
+            }
+          }
         })
       }
       // eslint-disable-next-line
     } catch (e: any) {
       if (
         e.code === 'CALL_EXCEPTION' &&
         Array.isArray(e.errorArgs) &&
-        e.errorArgs[0] === 'BAL#001'
+        (e.errorArgs[0] === 'BAL#001' || e.errorArgs[0] === 'BAL#526')
```

Comment:
could we add a comment here with referece to balancer's doc
```suggestion
                // ref: https://docs.balancer.fi/reference/contracts/error-codes.html
                (e.errorArgs[0] === 'BAL#001' || e.errorArgs[0] === 'BAL#526')
```